kvson.go

```diff
@@ -81,6 +81,21 @@ func (s *KVSON) Exists(key string) bool {
 	return false
 }
 
+func (s *KVSON) Delete(key string) bool {
+	if s.Exists(key) {
+		filename := filepath.Join(s.Path, key)
+		if _, err := os.Stat(filename); err == nil {
+
+			err := os.Remove(filename)
+			if err == nil {
```

Comment:
Golang antipattern: Return proper errors ( not booleans ) first. 
Instead of checking if the file exists to delete, to it backwards check if it does not exist first and return a proper error type.
If the file exists return nil error not true.